database/build/clean.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package build
+
+import (
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/database"
+	"github.com/go-vela/types/library"
+	"github.com/sirupsen/logrus"
+)
+
+// CleanBuilds updates builds to an error with a provided message with a created timestamp prior to a defined moment.
+func (e *engine) CleanBuilds(msg string, before int64) (int64, error) {
+	logrus.Tracef("cleaning pending or running builds in the database created prior to %d", before)
+
+	b := new(library.Build)
+	b.SetStatus(constants.StatusError)
+	b.SetError(msg)
```

Comment:
I think we should set the finished time too.

```suggestion
	b := new(library.Build)
	b.SetStatus(constants.StatusError)
	b.SetError(msg)
	b.SetFinished(time.Now().UTC().Unix())
```